index.py

```diff
@@ -91,9 +93,52 @@ def sort_app(organize_by, app_list):
                 for item in items:
                     organized_app_list[item].append(app)              
             else:
-                print("How did it even happen?")
+                # handling apps without Cat or Dev list.
+                organized_app_list['uncategorized'].append(app)
+
     return organized_app_list
 
+def get_git_url(module_name, app_name, git_commit_hash):
+    ''' Return url of git repository of the app.
+    Args: 
+        module_name: module name derived from app_id.
+        app_name: app name derived from app_id.
+        git_commit_hash: git commit hash of the app. 
+    Returns: 
+        Url of git repository.
+    '''
+
+    module_payload = {
+        'id': 0,
+        'method': 'Catalog.get_module_version',
+        'version': '1.1',
+        'params': [{   
+                        'module_name': module_name,
+                        'git_commit_hash' : git_commit_hash,
+                        'include_compilation_report': 1
+                    }]
+    }
+    
+    module_resp = requests.post(_catalog_url, data=json.dumps(module_payload))
+    try:
+        module_resp_json = module_resp.json()
+        # App info is stored in the first element of the result array.
+
+    except ValueError as err:
```

Comment:
Should this do any more failing? Does that need to propagate up to whatever calls `get_git_url`? Right now, if that fails (i.e. module_resp isn't really JSON), it'll just print the error and continue. Then line 130 will crash again, and say something about `module_resp_json` not being defined.